backend/src/main/java/com/bakdata/conquery/models/query/concept/specific/CQExternal.java

```diff
@@ -133,38 +132,32 @@ public void collectResultInfos(ResultInfoCollector collector) {
 		EVENT_DATE {
 			@Override
 			public BitMapCDateSet readDates(int[] dateIndices, String[] row) throws ParsingException {
-				return BitMapCDateSet.create(Collections.singleton(CDateRange.exactly(DateFormats.parseToLocalDate(row[dateIndices[0]]))));
+				return BitMapCDateSet.create(CDateRange.exactly(DateFormats.parseToLocalDate(row[dateIndices[0]])));
 			}
 		},
 		START_END_DATE {
 			@Override
 			public BitMapCDateSet readDates(int[] dateIndices, String[] row) throws ParsingException {
 				LocalDate start = row[dateIndices[0]] == null ? null : DateFormats.parseToLocalDate(row[dateIndices[0]]);
 				LocalDate end = (dateIndices.length < 2 || row[dateIndices[1]] == null) ?
-					null :
-								 DateFormats.parseToLocalDate(row[dateIndices[1]]);
+								null :
+								DateFormats.parseToLocalDate(row[dateIndices[1]]);
 
 				CDateRange range;
-				if (start != null && end != null) {
-					range = CDateRange.of(start, end);
-				}
-				else if (start != null) {
-					range = CDateRange.atLeast(start);
-				}
-				else if (end != null) {
-					range = CDateRange.atMost(end);
+				if (start == null && end == null) {
+					return null;
 				}
 				else {
-					return null;
+					range = CDateRange.of(start, end);
 				}
 
-				return BitMapCDateSet.create(Collections.singleton(range));
+				return BitMapCDateSet.create(range);
```

Comment:
Du kannst hier `return BitMapCDateSet.create(CDateRange.of(start, end));` machen. Das `of` kann mit null umgehen, so wie ich das sehe.